### PR TITLE
fix: prevent dragging when resizing

### LINKED ISSUE #12774: [Bug]-[500]:Table column border is sensitive and triggers both move and resize actions on mouse click leading to bad UX
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

Resizing table columns causes columns to get rearranged since the border is sensitive to both resize and rearrange action. This causes UX issues where user has to be slow while resizing as seen in the video and control is still on the selected column so either the column is resized or rearranged when user lets go of control by release click action.

### Steps To Reproduce

1. Go to a table widget and add a few columns
2. Try to re-size columns by hovering over the column boundary to see the resize icon.
3. Click to select the resize option and observe cursor change to move icon
4. Let go of cursor click at various points on the column header and observe that the column size readjusts to the point where the resize action is registered (see video attached at the 4 second mark to see this behaviour)

[LOOM DEMO](https://www.loom.com/share/70dd8d17372b4fd392b6b67d7948147f)

### Public Sample App

_No response_

### Version

Cloud

### PR DESCRIPTION
**Pull request by techbhavin**
## Description

> Prevent dragging on resizing  table column  on Mouse click action

Fixes #12774

## Type of change

- Bugfix (non-breaking change which fixes an issue)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:red_circle: Total coverage has decreased</summary>

    // Code coverage diff between base branch:release and head branch: fix/issue-12774 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :red_circle: | total | 56.59 **(-0.01)** | 38.37 **(-0.01)** | 35.86 **(0)** | 56.84 **(0)**
 :red_circle: | app/client/src/utils/WorkerUtil.ts | 88.98 **(-0.78)** | 68.63 **(-3.92)** | 100 **(0)** | 92.38 **(-0.95)**
 :red_circle: | app/client/src/widgets/TableWidget/component/index.tsx | 26.25 **(-1.02)** | 2.44 **(-0.06)** | 15.79 **(0)** | 25 **(-0.68)**</details>

### CODE REVIEW

File: app/client/src/widgets/TableWidget/component/index.tsx [hunk 1]
```diff
@@ -148,6 +148,11 @@ function ReactTableComponent(props: ReactTableComponentProps) {
       header.setAttribute("draggable", true);
 
       header.ondragstart = (e: React.DragEvent<HTMLDivElement>) => {
+        // check if table column is resizing
+        const isResizing = !!document.querySelectorAll(".resizer.isResizing")
```
Comment (jsartisan) [inline app/client/src/widgets/TableWidget/component/index.tsx:152]: This selector seems very generic. I remember using that class for resizor of entity explorer. Can't we use a redux state for resizing?

**Review by jsartisan — APPROVED** [review]

**Comment by chandannkumar** [comment]
- Tested this PR on windows & mac and seems smooth experience while resizing the columns on Table widget
- Prevent dragging on resizing table column on Mouse click action - Fixed

**Pull request merged**